packages/flutter/lib/src/rendering/semantics.dart

```diff
@@ -11,6 +11,8 @@ import 'package:flutter/painting.dart';
 import 'package:vector_math/vector_math_64.dart';
 
 import 'node.dart';
+import 'object.dart';
+import 'viewport.dart';
```

Comment:
we can't import viewport.dart here. That's a layering violation.

The rendering library has several layers. At the bottom is object.dart, layer.dart, node.dart, semantics.dart -- features that are coordinate system agnostic.

Below that we have RenderSliver, RenderBox, and the like, which know about coordinate systems but are agnostic to specific layout algorithms.

Below that we have things like flex.dart, proxy_box.dart, the various slivers, etc. They introduce specific opinions about layout and paint and so on.